### PR TITLE
Fix sighup handler [T2625]

### PR DESCRIPTION
**Pull request by Lagovas**
Found that process reloading on SIGHUP stopped properly handling for acra-server and acra-translator. Previously it used first argument of the process which is process's name. Under the hood golang calls [execve](https://pkg.go.dev/syscall#Exec) which doesn't search binary in [PATH](https://man7.org/linux/man-pages/man2/execve.2.html). Linux has [execlp(), execvp(), and execvpe()](https://linux.die.net/man/3/execlp) functions for that. But instead of using raw syscall's we use golang's wrappers from syscall package which doesn't provide these functions. Otherwise it has portability and do other useful things like [child checking, lock handling, etc](https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/syscall/exec_unix.go;l=141)

So, in this PR added searching absolute path of binary from binary name before calling ForkExec. Additionally added integration tests that checks sockets forwarding to child process and pulling config_file changes on SIGHUP.

## Checklist

- [X] Change is covered by automated tests
- [X] The [coding guidelines] are followed
- [ ] Public API has proper documentation in the [Acra documentation] site or has PR on [documentation repository] 
  with new changes
- [ ] CHANGELOG.md is updated (in case of notable or breaking changes)
- [ ] CHANGELOG_DEV.md is updated
- [X] Benchmark results are attached (if applicable)
- [X] [Example projects and code samples] are up-to-date (in case of API changes) 

[coding guidelines]: https://golang.org/doc/effective_go
[Example projects and code samples]: https://github.com/cossacklabs/acra-engineering-demo
[Acra documentation]: https://docs.cossacklabs.com/
[documentation repository]: https://github.com/cossacklabs/product-docs

### CODE REVIEW

File: cmd/acra-translator/acra-translator.go [hunk 1]
```diff
@@ -561,8 +562,14 @@ func realMain() error {
 		}
 		log.Debugf("Forking new process of %s", ServiceName)
 
+		binaryPath, err := exec.LookPath(os.Args[0])
```
[the remark this answers is not in the document]
Comment (Lagovas) [reply cmd/acra-translator/acra-translator.go:567]: no, we don't sure. Golang doesn't fork syscall  due to complicated runtime and fork limitations - https://stackoverflow.com/questions/28370646/how-do-i-fork-a-go-process
execve* requires path to executable that should be loaded. We try to find executable using same ENV variables used to run process itself. But yes, if we have `PATH=/dir1:/dir2` and `acra-server` was loaded from `/dir2/acra-server` and before sighup somebody place `acra-server` in `/dir1` then we will load new binary.

**Review by Zhaars — APPROVED** [review]
Everything is clear to me

**Review by G1gg1L3s — APPROVED** [review]
Cool!

**Review by G1gg1L3s — APPROVED** [review]
:+1:

**Pull request merged**